core/errorhelper.h

```diff
@@ -8,10 +8,18 @@
 extern "C" {
 #endif
 
+#ifdef __GNUC__
+#define __printf(x, y) __attribute__((__format__(__printf__, x, y)))
+#else
+#define __printf(x, y)
+#endif
+
 extern int verbose;
-extern int report_error(const char *fmt, ...);
+extern int __printf(1, 2) report_error(const char *fmt, ...);
+extern void __printf(1, 2) report_info(const char *fmt, ...);
 extern void set_error_cb(void(*cb)(char *));	// Callback takes ownership of passed string
```

Comment:
so now we have no report_error declaration anymore? My quick search seems to indicate that errorhelper.h is the only place where it was declared in a header.
What am I missing?

Reply:
It's not removed. The `__printf(1,2)` attribute is added so that we get compiler checls.